app/connector/kafka/src/main/resources/META-INF/syndesis/connector/kafka.json

```diff
@@ -153,18 +124,102 @@
         {
           "label": "TLS",
           "value": "SSL"
+        },
+        {
+          "label": "SASL_SSL",
+          "value": "SASL_SSL"
         }
       ],
       "group": "common,security",
       "javaType": "java.lang.String",
       "kind": "property",
       "label": "security",
-      "labelHint": "The transport protocol. Select TLS for encryption in transit.",
+      "labelHint": "The security protocol. Select TLS for encryption in transit. Select SASL_SSL when authenticating with SASL mechanism and SSL for transport.",
       "order": "2",
       "required": true,
       "secret": false,
       "type": "string"
     },
+    "saslMechanism": {
+      "componentProperty": true,
+      "deprecated": false,
+      "displayName": "SASL Mechanism",
+      "enum": [
+        {
+          "label": "PLAIN",
+          "value": "PLAIN"
+        },
+        {
+          "label": "OAUTHBEARER",
+          "value": "OAUTHBEARER"
+        }
+      ],
+      "group": "common,security",
+      "javaType": "java.lang.String",
+      "kind": "parameter",
+      "label": "security",
+      "labelHint": "The SASL Mechanism used.",
+      "order": "3",
+      "required": false,
+      "secret": false,
+      "type": "string"
+    },
+    "username": {
+      "componentProperty": false,
+      "deprecated": false,
+      "displayName": "Username",
+      "group": "common",
+      "javaType": "java.lang.String",
+      "kind": "parameter",
+      "label": "security",
+      "labelHint": "OAuth Client ID or the username",
+      "order": "4",
+      "required": false,
+      "secret": false,
+      "type": "string"
+    },
+    "password": {
+      "componentProperty": false,
+      "deprecated": false,
+      "displayName":  "Password",
+      "group": "common",
+      "javaType": "java.lang.String",
+      "kind": "parameter",
+      "label": "security",
+      "labelHint": "OAuth Client Secret or the password",
+      "order": "5",
+      "required": false,
+      "secret": true,
+      "type": "string"
+    },
+    "saslLoginCallbackHandlerClass": {
```

Comment:
Can we make this the default and not configurable by the user.

Reply:
Going to have a default one (as defined in the placeholder) if user selects oauthbearer, and having the field available if the user wants to set a different class from kafka clients.jar.